services/spar/src/Spar/SCIM.hs

```diff
@@ -303,23 +318,111 @@ instance SCIM.UserDB (ReaderT TeamId Spar) where
     maybe (error "How can there be no user?") (pure . toSCIMUser) =<<
       lift (getUser buid)
 
-  -- update   :: UserId -> User -> m StoredUser
-  -- patch    :: UserId -> m StoredUser
-  -- delete   :: UserId -> m Bool  -- ^ Return 'False' if the group didn't exist
-  -- getMeta  :: m Meta
+  update :: (TeamId, Maybe IdPId)
+         -> Text
+         -> SCIM.User.User
+         -> SCIM.SCIMHandler Spar SCIM.StoredUser
+  update = error "SCIM.User.update is not implemented yet"
+
+  delete :: (TeamId, Maybe IdPId) -> Text -> SCIM.SCIMHandler Spar Bool
+  delete = error "SCIM.User.delete is not implemented yet"
+
+  getMeta :: (TeamId, Maybe IdPId) -> SCIM.SCIMHandler Spar SCIM.Meta
+  getMeta = error "SCIM.User.getMeta is not implemented yet"
 
 ----------------------------------------------------------------------------
 -- GroupDB
 
-instance SCIM.GroupDB (ReaderT TeamId Spar) where
+instance SCIM.GroupDB Spar where
   -- TODO
 
 ----------------------------------------------------------------------------
 -- AuthDB
 
-instance SCIM.AuthDB (ReaderT TeamId Spar) where
-  -- TODO
+instance SCIM.AuthDB Spar where
+  type AuthData Spar = ScimToken
+  type AuthInfo Spar = (TeamId, Maybe IdPId)
+
   authCheck Nothing =
-    pure Unauthorized
-  authCheck (Just (SCIM.SCIMAuthData uuid _pass)) =
-    pure (Authorized (SCIM.Admin uuid))
+      SCIM.throwSCIM (SCIM.unauthorized "Token not provided")
+  authCheck (Just token) =
+      maybe (SCIM.throwSCIM (SCIM.unauthorized "Invalid token")) pure =<<
```

Comment:
Why does the new code use 4 spaces indentation while the old code uses 2 spaces?

Reply:
because it can?  :-)

we are thinking about using something like [brittany](http://hackage.haskell.org/package/brittany) to resolve all these discussions once and for all, but until that's ready for our code base, we try not to spend too much time on being too strict about it.

Reply:
I usually do the same indentation as the file already has, but Spar has a mix of indentation styles so sometimes I just go with what my editor suggests.